src/adagrams.js

```diff
@@ -1,15 +1,183 @@
+        'E',
+        'F',
+        'G',
+        'H',
+        'I',
+        'J',
+        'K',
+        'L',
+        'M',
+        'N',
+        'O',
+        'P',
+        'Q',
+        'R',
+        'S',
+        'T',
+        'U',
+        'V',
+        'W',
+        'X',
+        'Y',
+        'Z',
+    ];
+    const randomLetters = (choices) =>
+        choices[Math.floor(Math.random() * choices.length)];
+    let hand = [];
+    while (hand.length != 10) {
+        let letter = randomLetters(letters);
+        for (const [key, value] of Object.entries(letterPool)) {
+            if (letter == key && value != 0) {
+                letterPool[key] -= 1;
+                hand.push(letter);
+            }
+        }
+    }
+    return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+    input = input.toUpperCase();
+    const guessedLetters = Array.from(input);
+    let booleanList = [];
+    const letterCountGuessedLetters = guessedLetters.reduce((obj, letter) => {
+        //counts the number of letters in the input
+        const count = obj[letter] || 0; //if the letter is not in the object, it will be set to 0
+        return {...obj, [letter]: count + 1 }; //adds the letter to the object
+    }, {});
+    const letterCountLettersInHand = lettersInHand.reduce((obj, letter) => {
+        //counts the number of letters in the hand
+        const count = obj[letter] || 0; //if the letter is not in the object, it will be set to 0
+        return {...obj, [letter]: count + 1 }; //adds the letter to the object
+    }, {});
+    for (const [letter, count] of Object.entries(letterCountGuessedLetters)) {
+        //checks if the letter is in the hand
+        if (count <= letterCountLettersInHand[letter]) {
+            //if the letter is in the hand, true or false will be added to the boolean list
+            booleanList.push(true);
+        } else {
+            booleanList.push(false);
+        }
+    }
+    if (booleanList.every(Boolean)) {
+        //if all the booleans are true, return true
+        return true;
+    } else {
+        return false;
+    }
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+    const scoreChart = {
+        1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+        2: ['D', 'G'],
+        3: ['B', 'C', 'M', 'P'],
+        4: ['F', 'H', 'V', 'W', 'Y'],
+        5: ['K'],
+        8: ['J', 'X'],
+        10: ['Q', 'Z'],
+    };
+    word = word.toUpperCase();
+    let score = 0;
+    if (word === '') {
+        //if word is empty, return 0
+        return score;
+    }
+    for (const letter of word) {
+        //for each letter in word
+        for (const [points, letters] of Object.entries(scoreChart)) {
+            //for each letter in scoreChart
+            if (letters.includes(letter)) {
+                //if letter is in scoreChart
+                score += parseInt(points); //add points to score
```

Comment:
Nice approach! 

After we find the letter and increase the score, we could `break` out of the inner loop here to reduce unnecessary  iterations.